### PR TITLE
feat(mono/redm): initial works for rdr

### PR DESCRIPTION
**Pull request by AvarianKnight**
### Goal of this PR
Add same helper methods that FiveM has

### How is this PR achieving the goal
Start implementing base classes for RedM.

### This PR applies to the following area(s)
RedM, SCRT:C#

### Successfully tested on

**Game builds:** 1491

### Checklist

- [x] Code compiles and has been tested successfully.
- [x] Code explains itself well and/or is documented.
- [x] My commit message explains what the changes do and what they are for.
- [x] No extra compilation warnings are added by these changes.

### CODE REVIEW

File: code/client/clrcore-v2/Shared/Player.cs [hunk 1]
```diff
@@ -28,6 +28,9 @@ internal Player(ulong handle) : base(handle)
 		/// </summary>
 		public int Handle => (int)m_nativeValue;
 #endif
+
+		public static implicit operator CString(Player p) => p.Handle.ToString();
```
Comment (AvarianKnight) [inline code/client/clrcore-v2/Shared/Player.cs:32]: Just a note for me to remove this , it wasn't meant to make it into this PR

**Comment by AvarianKnight** [comment]
This PR should be good to review.

The Goal of this PR is to implement some base classes for RedM to allow players who want to use it to get started.

This change set also introduces a `UiFeed` class so the user can generate feed arguments

**Comment by AvarianKnight** [comment]
Some changes that were made:

`Model` implements `IDisposable` so it will be disposed of properly when the object goes out of scope
`Vehicle` has more stuff implemented

Given recent events I likely wont work on this anymore so anyone can feel free to continue this PR if it needs changes.

File: code/client/clrcore-v2/Game/Shared/Game.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			}
+
+			string entType = s_entityTypeMap[typeof(T)];
+
+			// Trying to use .Cast here won't work, so we use opt to use the converter
+			return (Natives.GetGamePool(entType) as object[]).Select(entObj => {
+				int entId = Convert.ToInt32(entObj);
+				var data = Activator.CreateInstance(typeof(T), entId);
+				return data as T;
+			});
```
Comment (AvarianKnight) [inline code/client/clrcore-v2/Game/Shared/Game.cs:79]: This needs a wrapper for mono v1

**Review by FabianTerhorst — APPROVED** [review]
Thanks a lot for the contribution. Seems to work fine 👍

**Pull request merged**